scipy/special/_basic.py

```diff
@@ -2953,6 +2956,137 @@ def factorialk(n, k, exact=True):
         raise NotImplementedError
 
 
+def stirling2(N, K, exact=True):
+    """Generate Stirling number(s) of the second kind.
+
+        Stirling numbers of the second kind count the number of ways to
+        partition a set with N elements into K non-empty subsets.
+
+        The values this function returns are calculated using a dynamic
+        program which avoids redundant computation across the subproblems
+        in the solution.
+
+        The numbers are sometimes denoted
+        .. math::
+
+            `{N \brace K}`;
+
+        see [GKP]_ for details. This is often expressed-verbally-as
+        "N subset K".
+
+        Parameters
+        ----------
+        N : int, ndarray
+            Number of things.
+        K : int, ndarray
+            Number of non-empty subsets taken.
+        exact : bool, optional
+            For integers, if `exact` is False, then floating point precision is
+            used, otherwise the result is computed exactly. For non-integers,
+            if `exact` is True, is truncated. Defaults to True.
+
+        Returns
+        -------
+        val : int, float, ndarray
+            The number of partitions.
+
+        See Also
+        --------
+        comb : The number of combinations of N things taken k at a time.
+
+        Notes
+        -----
+        - If N < 0, or K < 0, then 0 is returned.
+        - If K > N, then 0 is returned.
+        The output type will always be `int` or ndarray of `object`.
+        The input must contain either numpy or python integers otherwise a
+        TypeError is raised.
+
+        References
+        ----------
+         .. [GKP] R. L. Graham, D. E. Knuth and O. Patashnik, “Concrete
+        Mathematics: A Foundation for Computer Science,” Addison-Wesley
+        Publishing Company, Boston, 1989. Chapter 6, page 258.
```

Comment:
Oh sorry, I missed the lint failure.

```suggestion
         .. [GKP] R. L. Graham, D. E. Knuth and O. Patashnik, "Concrete
        Mathematics: A Foundation for Computer Science," Addison-Wesley
        Publishing Company, Boston, 1989. Chapter 6, page 258.
```